### PR TITLE
Encapsulate presentation in a presenter

### PR DESCRIPTION
**Pull request by laurentqro**

### CODE REVIEW

File: lib/models/office.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
 class Office
-  @offices = ["london", "madison"]
+  LOCATIONS = ["london", "madison"].freeze
```
Comment (fabientownsend) [inline lib/models/office.rb]: 👍

File: lib/commands/crafter/add_office.rb [hunk 2]
```diff
@@ -25,21 +30,25 @@ def updated_apprentice
...
+          )
         end
 
-        "You were added to the office: #{@office}"
+        "You were added to the office: #{@office.location}"
       elsif @crafter || @apprentice
-        Logger.info("#{@crafter.user_name} use a unavailable office: #{@office}")
-        "The office available are: #{Office.list.map(&:capitalize).join(", ")}"
+        Logger.info("#{@crafter.user_name} used an unavailable office: #{@office.location}")
+        "The offices available are: #{OfficePresenter.locations}"
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: Couldn't the presentation be inline here? I am not sure of the value provided by the presenter as it is not used anywhere else 
```suggestion
        "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
```

File: lib/commands/crafter/add_office.rb [hunk 3]
```diff
@@ -25,21 +30,25 @@ def updated_apprentice
     end
 
     def run
-      if Office.available?(@office) && (@crafter || @apprentice)
+      if @office.available? && (@crafter || @apprentice)
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: We could remove this condition as we are doing the verification after

```suggestion
      if @office.available?
```

or have
```
if @office.available? && @crafter
...
end
```

```
if @office.available? && @apprentice
...
end
```

File: lib/commands/crafter/add_office.rb [hunk 4]
```diff
@@ -25,21 +30,25 @@ def updated_apprentice
...
+          Logger.info(
+            "#{@office.location} was added to #{@crafter.user_name} - #{updated_crafter}"
+          )
         end
 
         if @apprentice
-          @apprentice.add_office(@office)
-          Logger.info("#{@office} was added to #{@apprentice.user_name} - #{updated_apprentice}")
+          @apprentice.add_office(@office.location)
+          Logger.info(
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: I believe that we can delete these event log

Comment (fabientownsend) [reply lib/commands/crafter/add_office.rb]: I mean, the `#{update_apprentice}`, likewise the `updated_crafter`

File: lib/commands/crafter/add_office.rb (hunk 2 above)
Comment (laurentqro) [reply lib/commands/crafter/add_office.rb]: yeah it might be me optimising too early :)

**Pull request merged**